mautic/mautic
# Refactor Offline.php to use twig templates

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

Since we are moving away from PhpEngine to Twig, we have to remove any reference to hardcoded PHP code that renders pages. Offline.php is one of those pages. 

This PR removes the file and makes it into a twig template, it is refactoring the old code.

**I prefer someone with twig experience to validate the twig template as I am not the best at this **

Here is an example of how the page looks like after converting to twig, I noticed that the error below is a special case it always show this string
![image](https://user-images.githubusercontent.com/12627658/226877041-53410a02-42be-439e-b26d-6d5875661975.png)




#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Break Mautic, either add invalid code, or change the database connection by changing `local.php` and provide with wrong database connection
3. Check that you can see a similar offline message as the one in the screenshot
4. Try to raise an exception to see that error 500 page works as expected
5. Make a syntax error in `local.php` in production you should see something similar to 
![image](https://user-images.githubusercontent.com/12627658/226885898-b1316003-34ff-45ba-8a81-0f68d6afba4c.png)
This is the same error under `dev` environment 
![image](https://user-images.githubusercontent.com/12627658/226886565-46643e51-ccfc-49e4-b4b0-32c2c18439b0.png)
